stl/inc/span

```diff
@@ -0,0 +1,685 @@
+// span standard header
+
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#pragma once
+#ifndef _SPAN_
+#define _SPAN_
+#include <yvals_core.h>
+#if _STL_COMPILER_PREPROCESSOR
+
+#if _HAS_CXX20
+#include <array>
+#include <cstddef>
+#include <type_traits>
+
+#pragma pack(push, _CRT_PACKING)
+#pragma warning(push, _STL_WARNING_LEVEL)
+#pragma warning(disable : _STL_DISABLED_WARNINGS)
+_STL_DISABLE_CLANG_WARNINGS
+#pragma push_macro("new")
+#undef new
+
+_STD_BEGIN
+
+inline constexpr size_t dynamic_extent = static_cast<size_t>(-1);
+
+// STRUCT TEMPLATE _Span_iterator
+template <class _Ty>
+struct _Span_iterator {
+#ifdef __cpp_lib_concepts
+    using iterator_concept = contiguous_iterator_tag;
+#endif // __cpp_lib_concepts
+    using iterator_category = random_access_iterator_tag;
+    using value_type        = remove_cv_t<_Ty>;
+    using difference_type   = ptrdiff_t;
+    using pointer           = _Ty*;
+    using reference         = _Ty&;
+
+    constexpr _Span_iterator() noexcept = default;
+
+#if _ITERATOR_DEBUG_LEVEL >= 1
+    constexpr _Span_iterator(const pointer _Pointer, const pointer _Begin, const pointer _End) noexcept
+        : _Myptr(_Pointer), _Mybegin(_Begin), _Myend(_End) {}
+#else // ^^^ _ITERATOR_DEBUG_LEVEL >= 1 ^^^ // vvv _ITERATOR_DEBUG_LEVEL == 0 vvv
+    constexpr explicit _Span_iterator(const pointer _Pointer) noexcept : _Myptr(_Pointer) {}
+#endif // _ITERATOR_DEBUG_LEVEL
+
+    template <class _OtherType, enable_if_t<is_convertible_v<remove_pointer_t<_OtherType> (*)[], _Ty (*)[]>, int> = 0>

```

Comment:
`remove_pointer_t<_OtherType>` seems to be bogus. Assuming that's removed, this is a bit more permissive than "`iterator` implicitly converts to `const_iterator`" that's necessary to meet the requirements for container iterators. I'm not necessarily opposed - `is_convertible_v<_OtherType (*)[], _Ty (*)[]>` is definitely the right requirement - just pointing out the difference.

Reply:
Yep, that was obviously a test to check your vigilance and not me screwing up copy & paste

What I like about it is that it mirrors the requirements from the array/container constructors which makes it a bit more consistent with the rest of the code

Fixed